src/cares.c

```diff
@@ -486,16 +490,37 @@ query_txt_cb(void *arg, int status,int timeouts, unsigned char *answer_buf, int
         goto callback;
     }
 
+    first_record = 1;
```

Comment:
You should be able to not use a variable for this. Store a pointer to the current object (you already have a tmp_obj). If you get a record start and it's not NULL, append it to the list and create a new one. Same applies to the cffi changes.